File: exercises/Ex03CaseMatch/case-match.js

const app = require("express")()
const bodyParser = require("body-parser")

app.use(bodyParser.raw({ type: "*/*" }))

app.post("/case", (req, res) => {
  let result

  switch (req.body.toString()) {
    case "1":
      result = "one"
      break
    case "2":
      result = "two"
      break
    case "3":
      result = "three"
      break
    default:
      result = "What am I, a mathematician?"
  }

  res.send(result)
})

app.listen(7879)

File: exercises/Ex03CaseMatch/CaseMatch.hs
module Ex03CaseMatch.CaseMatch where

import qualified Zero.Server as Server


caseHandler :: Server.Request -> Server.Response
caseHandler req
  = Server.stringResponse res
  where
    res
      = case Server.requestBody req of
          "1" -> "one"
          "2" -> "two"
          "3" -> "three"
          _   -> "What am I, a mathematician?"

main :: IO ()
main
  = Server.startServer
      [ Server.simpleHandler Server.POST "/case" caseHandler
      ]
